docs/Developers-Guide.md

```diff
@@ -77,21 +78,83 @@ Similarly, if code does not get accepted upstream by the main developers, it is
 # Development Environment
 
 ## Installation
-Developers might want to install IntelMQ with `pip3 -e`, which gives you a so called *editable* installation. No code is copied in the libraries directories, there's just a link to your code.
 
-    pip3 install -e .
+Developers MUST have a fork repository of IntelMQ in order to commit the new code to their repository and then be able to do pull requests to main repository.
+
+The following instructions will use a so called *editable* installation. No code is copied in the libraries directories, there's just a link to your code. However, configuration files still required to be moved to `/opt/intelmq` as the instructions show.
+
+```bash
+sudo -s
+
+git clone https://github.com/<your username>/intelmq.git /opt/dev_intelmq
+cd /opt/dev_intelmq
+
+pip3 install -e .
+
+useradd -d /opt/intelmq -U -s /bin/bash intelmq
+
+mkdir /opt/intelmq
+mkdir -p /opt/intelmq/var/lib/bots/file-output/
+mkdir -p /opt/intelmq/var/log/
+
+cp -R /opt/dev_intelmq/intelmq/etc /opt/intelmq/
+cp -R /opt/dev_intelmq/intelmq/bots/BOTS /opt/intelmq/etc/
+
+chmod -R 0770 /opt/intelmq
+chown -R intelmq.intelmq /opt/intelmq
+```
+
+**Note:** please do not forget that configuration files, log files will be available on `/opt/intelmq`. However, if your development is somehow related to any configuration file, keep using `/opt/intelmq` and then, before commit, change the configurations files on `/opt/dev_intelmq/intelmq/etc/` with your changes on `/opt/intelmq/etc/`.
+
+
+## How to develop
+
+After you successfully setup your IntelMQ development environment, you can perform any development on any `.py` file on `/opt/dev_intelmq`. After you change, you can use the normal procedure to run the bots:
+
+```bash
+su - intelmq
+
+intelmqctl start spamhaus-drop-collector
+
+tail -f /opt/intelmq/var/log/spamhaus-drop-collector.log
+```
+
+You can also add new bots, creating the new `.py` file on the proper directory inside `cd /opt/dev_intelmq/intelmq`. However, your IntelMQ installation with pip3 needs to be updated. Please check the following section.
+
 
 ## Update
 
-If you do any changes on setup.py, data files (e.g. example configurations) or add new bots, you need to rerun the installation routine.
+In case you developed a new bot, you need to update your current development installation. In order to do that, please follow this procedure:
+
+
+1. Add the new bot information to `/opt/dev_intelmq/intelmq/bots/BOTS`, not `/opt/intelmq/etc/BOTS`.
+2. Make sure that you have your new bot in the right place and the information on BOTS file is correct.
+3. Execute the following commands:
+
+```bash
+sudo -#!/bin/sh
+
+cd /opt/dev_intelmq
+pip3 install --upgrade -e .
```

Comment:
with `--upgrade` that upgrades all the dependencies too. Without it, only intelmq will be reinstalled/upgraded

Reply:
fixed